### PR TITLE
[ENG-29122] Support disable ddl transaction

### PR DESCRIPTION
**Pull request by zreisman**
Adds support for disable_ddl_transaction! so that we can run long running data migrations if needed.

### CODE REVIEW

File: lib/seed_migration/migration.rb [hunk 1]
```diff
@@ -6,6 +6,20 @@ module SeedMigration
...
+      # Disable the transaction wrapping this migration.
+      # You can still create your own transactions even after
+      # calling #disable_ddl_transaction!
+      def disable_ddl_transaction!
+        @disable_ddl_transaction = true
+      end
+    end
+
+    def disable_ddl_transaction
+      self.class.disable_ddl_transaction
```
Comment (bdurand) [inline lib/seed_migration/migration.rb]: Missing the `!` suffix on the method call here.

**Review by bdurand — CHANGES_REQUESTED** [review]
I think the method names should be changed to `disable_transaction!` since we aren't doing any DDL changes in data migrations. It's not consistent with the syntax for database migrations, but I think it makes more sense and removes a potential confusion about what's going on.

File: lib/seed_migration/migration.rb (hunk 1 above)
Comment (zreisman) [reply lib/seed_migration/migration.rb]: Supposed to. Just an attr accessor.  The bang is to set the var

**Comment by zreisman** [comment]
@bdurand Makes sense. Will do

File: lib/seed_migration/migration.rb [hunk 2]
```diff
@@ -6,6 +6,20 @@ module SeedMigration
...
+      # Disable the transaction wrapping this migration.
+      # You can still create your own transactions even after
+      # calling #disable_transaction!
+      def disable_transaction!
+        @disable_transaction = true
+      end
+    end
+
+    def disable_transaction
+      self.class.disable_transaction
```
Comment (bdurand) [inline lib/seed_migration/migration.rb]:
```suggestion
      self.class.disable_transaction!
```

**Review by bdurand — CHANGES_REQUESTED** [review]

File: lib/seed_migration/migration.rb (hunk 2 above)
Comment (zreisman) [reply lib/seed_migration/migration.rb]: The macro is only needed as a class method. This is to access the variable, not to change it. I renamed the variable to differentiate them.

**Review by bdurand — APPROVED** [review]

**Pull request merged**